src/python/pants/engine/legacy/graph.py

```diff
@@ -576,7 +575,7 @@ def owns_any_source(legacy_target: HydratedTarget) -> bool:
         to_visit.extend(tht.dependencies)
 
     return TransitiveHydratedTargets(
-        tuple(tht.root for tht in transitive_hydrated_targets), closure
+        tuple(tht.root for tht in transitive_hydrated_targets), FrozenOrderedSet(closure)
```

Comment:
This does have a minor performance hit to convert from `OrderedSet` to `FrozenOrderedSet`. I looked at refactoring this code to use a comprehension so that we can use `FrozenOrderedSet`, but it doesn't seem possible with the `while` loop.

We could choose to have `TransitiveHydratedTargets.closure` use `OrderedSet` instead of `FrozenOrderedSet`, but, from the start, the entire purpose of these `ordered_set` changes has been for this particular use case to make THTs safer..So, I think it's worth it to get the safety of immutablity + hashability.

Reply:
> So, I think it's worth it to get the safety of immutablity + hashability.

100% agreed.